src/app/api/schedule/[id]/route.ts

```diff
@@ -0,0 +1,84 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import { eq } from "drizzle-orm";
+
+import { db } from "@/lib/db";
+import { gameSessionSchedules } from "@/lib/db/schema";
+
+export async function GET(
+  req: NextRequest,
+  { params }: { params: { id: number } },
+) {
+  try {
+    const { id } = params;
+
+    const gameSessionSchedule = await db.query.gameSessionSchedules.findFirst({
+      where: eq(gameSessionSchedules.id, id),
+    });
+
+    if (!gameSessionSchedule) {
+      return new Response(`No GameSessionSchedule found for id: ${id}`, {
+        status: 404,
+      });
+    }
+
+    return NextResponse.json(gameSessionSchedule);
+  } catch {
+    return new Response("Internal Server Error", { status: 500 });
+  }
+}
+
+export async function DELETE(
+  req: NextRequest,
+  { params }: { params: { id: number } },
+) {
+  try {
+    const { id } = params;
+
+    const gameSessionSchedule = await db.query.gameSessionSchedules.findFirst({
+      where: eq(gameSessionSchedules.id, id),
+    });
+
+    if (!gameSessionSchedule) {
+      return new Response(`No GameSessionSchedule found for id: ${id}`, {
+        status: 404,
+      });
+    }
+
+    await db
+      .delete(gameSessionSchedules)
+      .where(eq(gameSessionSchedules.id, id));
+    return new Response(null, { status: 204 });
+  } catch {
+    return new Response("Internal Server Error", { status: 500 });
+  }
+}
+
+export async function PUT(
+  req: NextRequest,
+  { params }: { params: { id: number } },
+) {
+  try {
+    const { id } = params;
+
+    const updatedGameSession = await req.json();
```

Comment:
Same for this -- validate this schema

Reply:
I understand why we would need a schema for the POST request but for PUT, we dont know if the api will recieve a request that include all fields of the schema right?